### PR TITLE
feat(FEC-12611): Kaltura Image Entry Support - Load Kaltura Image Entries with or without KS

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

solves  FEC-12611

related prs: 
https://github.com/kaltura/kaltura-player-js/pull/598
https://github.com/kaltura/playkit-js/pull/697
https://github.com/kaltura/image-player/pull/1
https://github.com/kaltura/playkit-js-ui/pull/718

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/entities/image-source.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+   * @member - media source mimetype
+   * @type {string}
+   */
+  mimetype: string;
+
+  constructor(entry: Object) {
+    this.id = entry.id;
+    this.url = entry.dataUrl;
+    this.mimetype = 'image/jpeg';
+    this.templateUrl = ImageSource.createTemplateUrl(entry.dataUrl);
```
Comment (SivanA-Kaltura) [inline src/entities/image-source.js:29]: what does this do ?

**Comment by SivanA-Kaltura** [comment]
how does the player know what to how to play the image source ? do you need to make changes in playkit-js as well ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/entities/image-source.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  /**
+   * @member - media source mimetype
+   * @type {string}
+   */
+  mimetype: string;
+
+  constructor(entry: Object) {
+    this.id = entry.id;
+    this.url = ImageSource.extractBaseThumbnailUrl(entry.dataUrl);
+    this.mimetype = 'image/jpeg';
```
Comment (SivanA-Kaltura) [inline src/entities/image-source.js:25]: can't it be a different image format ? e.g. png

File: src/entities/image-source.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  /**
+   * Convert url to template url.
+   * @param {string} url - dataUrl.
+   * @returns {string} - The template url .
+   */
+  static extractBaseThumbnailUrl(url: string): string {
```
Comment (SivanA-Kaltura) [inline src/entities/image-source.js:33]: why is it static ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

**Comment by SivanA-Kaltura** [comment]
@JonathanTGold some basic unit tests i can think of:
- check that the parser handles an image response and adds the data to the image array
- check what happens when an entry has a poster vs an entry that doesn't have a poster

File: src/entities/image-source.js (hunk 3 above)
Comment (JonathanTGold) [reply src/entities/image-source.js:33]: as long as the method is instance agnostic (and the indication is that you don’t see the `this` keyword) it should be static

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**